pkg/app/api/grpcapi/piped_api.go

```diff
@@ -509,9 +513,68 @@ func (a *PipedAPI) ReportDeploymentStatusChanged(ctx context.Context, req *piped
 	if err = a.updateDeploymentRefStatusIfNecessary(ctx, req.DeploymentChainId, req.DeploymentChainBlockIndex, req.DeploymentId, req.Status, req.StatusReason); err != nil {
 		return nil, status.Error(codes.Internal, "unable to update deployment ref status of the deployment chain this deployment belongs to")
 	}
+
+	if err = a.sendCancelDeploymentCommandIfNecessary(ctx, pipedID, req.DeploymentId, req.DeploymentChainId, req.DeploymentChainBlockIndex); err != nil {
+		return nil, err
+	}
```

Comment:
This shouldn't be called the deployment has been transited to `ROLLING_BACK` status?

Reply:
Definitely not, nice catch 👍

Reply:
Updated by https://github.com/pipe-cd/pipe/pull/2893/commits/fa88c94ebd2128896a7d75bace99833b7420255e 🙏